src/devices/video/topcat.h

```diff
@@ -84,20 +85,36 @@ class topcat_device : public device_t
 	};
 
 	void window_move(void);
-	void execute_rule(uint16_t src, replacement_rule_t rule, uint16_t *dst);
+	void execute_rule(bool src, replacement_rule_t rule, bool *dst);
```

Comment:
does this actually treat `dst` as an array or allow `nullptr`?  If not it should be a reference, not a pointer.